framework/decode/file_processor.cpp

```diff
@@ -76,32 +80,62 @@ void FileProcessor::WaitDecodersIdle()
 
 bool FileProcessor::Initialize(const std::string& filename)
 {
-    bool success = false;
+    bool success = OpenFile(filename);
 
-    int32_t result = util::platform::FileOpen(&file_descriptor_, filename.c_str(), "rb");
+    if (success)
+    {
+        success = SetActiveFile(filename, true);
+        success = success && ProcessFileHeader();
+    }
+    else
+    {
+        GFXRECON_LOG_ERROR("Failed to open file %s", filename.c_str());
+        error_state_ = kErrorOpeningFile;
+    }
 
-    if ((result == 0) && (file_descriptor_ != nullptr))
+    // Find absolute path of capture file
+    if (success)
     {
-        success = ProcessFileHeader();
+        size_t last_slash_pos = filename.find_last_of("\\/");
+        if (last_slash_pos != std::string::npos)
+        {
+            absolute_path_ = filename.substr(0, last_slash_pos + 1);
+        }
+    }
 
-        if (success)
+    return success;
+}
+
+std::string FileProcessor::ApplyAbsolutePath(const std::string& file)
+{
+    if (absolute_path_.empty())
```

Comment:
Can you use `gfxrecon::util::filepath::Join` for this?  I don't think you'd even need to check `absoluate_path_.empty()`.